tempesta_fw/http_sess.c

```diff
@@ -318,22 +318,33 @@ search_cookie(TfwStickyCookie *sticky, const TfwStr *cookie, TfwStr *val,
 static int
 tfw_http_sticky_get_req(TfwHttpReq *req, TfwStr *cookie_val)
```

Comment:
What was the sense to crash on duplicate header in `__h2_msg_hdr_val()`? If the duplicated header doesn't lead to later crash and/or memory corruption, then please replace it with `WARN_ON_ONCE()`.

Reply:
The `__h2_msg_hdr_val()`, `__http_msg_hdr_val()` and `tfw_http_msg_clnthdr_val()` functions
can be called only against special headers. Since we block messages with duplicated singular special headers at early parsing stage, we assume that singular header can be processed by direct access, while non-singular require `TFW_STR_FOR_EACH_DUP` loops.

We've got a crash here, because `cookie` header is the only exception in h2 and the h2 RFC explicitly allow to duplicate `cookie` header.

I don't sure we really can replace `BUG_ON(TFW_STR_DUP(hdr))` to `WARN_ON()` for every header and string read and write operation.   Either we throw a error and check return code in all such operations, which are very frequent, or we use a kind of save fallback - first duplicate or similar solution. I don't like the later case, since attackers will be able to force us to skip header processing by adding empty header before used one, and processing may differ for duplicated and compounded (united with `,`) headers.

I've checked all the singular spec headers, and we have no issues while working with them, but we have some with nonsingular ones: #1403